lib/modules/manager/terraform/extract.spec.ts

```diff
@@ -619,6 +619,13 @@ describe('modules/manager/terraform/extract', () => {
           depType: 'helm_release',
           registryUrls: [undefined],
         },
+        {
+          currentValue: 'v0.22.1',
+          datasource: 'docker',
+          depName: 'public.ecr.aws/karpenter/karpenter',
+          depType: 'helm_release',
+          registryUrls: [undefined],
```

Comment:
@secustor do you know why we've `undefined` as registry url here? looks wrong too me, but isn't caused by this PR

Reply:
It simply wraps the value of `helmRelease.repository` in an array. I think this has been introduced in the initial implementation and I have kept it during the refactor to do not the changes.    

The `terraform` typing refactor removes this. https://github.com/renovatebot/renovate/pull/19815/files#diff-f699f41b656f879ddc9f56694fc56e6d6e0921f07a84d8543593ebbe7e7fd7e7L597

Reply:
@secustor Can you create a separate PR to fix that? looks like a bug to me, or is that only happen inside tests?

Reply:
Resolved in #19905.